core/workflow-activiti/src/main/java/org/apache/syncope/core/workflow/activiti/task/Recertify.java

```diff
@@ -0,0 +1,102 @@
+import org.apache.syncope.core.persistence.api.dao.UserDAO;
+import org.apache.syncope.core.persistence.api.entity.AnyUtilsFactory;
+import org.apache.syncope.core.persistence.api.entity.EntityFactory;
+import org.apache.syncope.core.persistence.api.entity.PlainSchema;
+import org.apache.syncope.core.persistence.api.entity.user.UPlainAttr;
+import org.apache.syncope.core.persistence.api.entity.user.User;
+import org.apache.syncope.core.workflow.activiti.ActivitiUserWorkflowAdapter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class Recertify extends AbstractActivitiServiceTask {
+
+    public static final String LAST_RECERTIFICATOR_ATTR = "lastRecertificator";
+    
+    public static final String LAST_RECERTIFICATION_ATTR = "lastRecertification";
+    
+    public static final String RECERTIFICATION_TIME = "identity.recertification.day.interval";
+
+    @Autowired
+    protected UserDAO userDAO;
+
+    @Autowired
+    protected EntityFactory entityFactory;
+
+    @Autowired
+    private PlainSchemaDAO plainSchemaDAO;
+
+    @Autowired
+    protected AnyUtilsFactory anyUtilsFactory;
+
+    private PlainSchema lastRecertificationSchema;
+
+    private PlainSchema lastRecertificatorSchema;
+
+    @Override
+    protected void doExecute(final String executionId) {
+
+        LOG.debug("Processing Recertification {}", executionId);
+        User user = engine.getRuntimeService().
+                getVariable(executionId, ActivitiUserWorkflowAdapter.USER, User.class);
+        String submitter = engine.getRuntimeService().
+                getVariable(executionId, ActivitiUserWorkflowAdapter.FORM_SUBMITTER, String.class);
+
+        LOG.debug("Saving Recertification information for user {}", user.getUsername());
+
+        lastRecertificatorSchema = plainSchemaDAO.find(LAST_RECERTIFICATOR_ATTR);
+        lastRecertificationSchema = plainSchemaDAO.find(LAST_RECERTIFICATION_ATTR);
+        setSchemaReadonly(false);
+
+
+        UPlainAttr recertifierAttr = entityFactory.newEntity(UPlainAttr.class);
+        recertifierAttr.setOwner(user);
+        recertifierAttr.setSchema(lastRecertificatorSchema);
+
+        recertifierAttr.add(submitter, anyUtilsFactory.getInstance(AnyTypeKind.USER));
+
+        DateFormat format = new SimpleDateFormat(lastRecertificationSchema.getConversionPattern());
+        String recertificationDate = format.format(new Date());
+        UPlainAttr recertAttr = entityFactory.newEntity(UPlainAttr.class);
+        recertAttr.setOwner(user);
+        recertAttr.setSchema(lastRecertificationSchema);
+
+        recertAttr.add(recertificationDate, anyUtilsFactory.getInstance(AnyTypeKind.USER));
+
+        user.add(recertAttr);
+        user.add(recertifierAttr);
+
+        setSchemaReadonly(true);
+
+        userDAO.save(user);
+    }
+
+    private void setSchemaReadonly(final boolean readonly) {
```

Comment:
This method can be removed for the same reasons stated above for static fields; however, it wasn't needed anyway as read-only attributes are meant for being managed via DAO rather than via DataBider.